helix-core/src/main/java/org/apache/helix/tools/ClusterSetup.java

```diff
@@ -135,37 +138,74 @@
   public static final String removeConstraint = "removeConstraint";
 
   private static final Logger _logger = LoggerFactory.getLogger(ClusterSetup.class);
-  private final String _zkServerAddress;
-  private final HelixZkClient _zkClient;
-  // true if ZkBaseDataAccessor was instantiated with a HelixZkClient, false otherwise
+  private String _zkServerAddress;
```

Comment:
It seems this field can still be final. Why do we need to remove the final keyword?

Reply:
If the user wants to create this class on multi-realm mode, FederatedZkClient will be used. FederatedZkClient does not and should not expose ZkAddress because it's an internal state info.

In fact, this `_zkServerAddress` was not being used anywhere and doesn't need to exist. I've removed this field from this class.